dali/kernels/imgproc/resample/resampling_setup.cc

```diff
@@ -98,80 +100,196 @@ SeparableResamplingSetup<spatial_ndim>::ComputeScaleAndROI(
   return roi;
 }
 
+template <int ndim>
+struct ProcessingOrderCalculator {
+  ProcessingOrderCalculator(ivec<ndim> in_size, ivec<ndim> out_size, ivec<ndim> filter_support)
+  : in_size(in_size), out_size(out_size), filter_support(filter_support) {
+  }
+
+  ProcessingOrder<ndim> operator()() {
+    for (int i = 0; i < ndim; i++)
+      best_order[i] = i;
+    min_cost = 1e+30f;
+    axis_mask = {};
+    curr_size = in_size;
+    run(0);
+    return best_order;
+  }
+
+ private:
+  float compute_cost_mul(int axis) {
+    // y-axis is likely to be the cheapest
+    return axis == 0 ? 1.4f : axis > 1 ? 1.2f : 1.0f;
+  }
+
+  static constexpr float size_bias = 3;
+
+  float calculate_cost(int pass, int axis) {
+    auto vol = volume(curr_size);
+    float base_compute_cost = filter_support[axis] * vol;
+    float compute_cost = compute_cost_mul(axis) * base_compute_cost;
+    return compute_cost + vol * size_bias;
+  }
+
+  void run(int pass, float total_cost = 0) {
+    if (total_cost >= min_cost)
+      return;
+
+    if (pass == ndim) {
+      min_cost = total_cost;
+      best_order = curr_order;
+    } else {
+      for (int a = 0; a < ndim; a++) {
+        if (axis_mask[a])
+          continue;
+        axis_mask[a] = true;
+        auto prev_size = curr_size[a];
+        curr_size[a] = out_size[a];
+        curr_order[pass] = a;
+
+        float pass_cost = calculate_cost(pass, a);
+        run(pass + 1, total_cost + pass_cost);
+
+        curr_size[a] = prev_size;
+        axis_mask[a] = false;
+      }
+    }
+  }
+
+  float min_cost;
+  ivec<ndim> in_size, out_size, curr_size, filter_support;
+  ProcessingOrder<ndim> best_order, curr_order;
+  bvec<ndim> axis_mask;
+};
+
+template <int ndim>
+ProcessingOrder<ndim> GetProcessingOrder(
+    ivec<ndim> in_size,
+    ivec<ndim> out_size,
+    ivec<ndim> filter_support) {
+  ProcessingOrderCalculator<ndim> poc(in_size, out_size, filter_support);
+  return poc();
+}
+
+template <>
+void SeparableResamplingSetup<2>::ComputeBlockLayout(SampleDesc &sample) {
+  for (int pass = 0; pass < 2; pass++) {
+    int axis = sample.order[pass];
+    // Horizontal pass (axis == 0) is processed in vertical slices
```

Comment:
I think we agreed to
```
/*
 */
```

Reply:
Only for doxygen.

Reply:
Yes, only for doxygen, and it was to have one doxygen syntax used throughout the codebase. One benefit of this is I don't have to reconfigure my addon that pregenerates the doxygen for given function for different parts of DALI. Which one is it doesn't really matter that much.